internal/compiler/llr/lower_to_item_tree.rs

```diff
@@ -203,20 +203,26 @@ fn lower_sub_component(
 
     if let Some(parent) = component.parent_element.upgrade() {
         // Add properties for the model data and index
-        if parent.borrow().repeated.as_ref().map_or(false, |x| !x.is_conditional_element) {
-            sub_component.properties.push(Property {
-                name: "model_data".into(),
-                ty: crate::expression_tree::Expression::RepeaterModelReference {
-                    element: component.parent_element.clone(),
-                }
-                .ty(),
-                ..Property::default()
-            });
-            sub_component.properties.push(Property {
-                name: "model_index".into(),
-                ty: Type::Int32,
-                ..Property::default()
-            });
+        match parent.borrow().repeated {
+            Some(RepeatedElementInfo::Repeater(_)) => {
```

Comment:
This needs the same check for conditional elements as in the original code, I think. Otherwise behavior changes.